pipeline/pipeline/resource.py

```diff
@@ -86,14 +102,18 @@ def __init__(self, source, root, **values):
         self._output_paths = set()
 
         for name, resource in values.items():
+            assert isinstance(resource, ResourceFile)
```

Comment:
In the spirit of the above comment; should this ever be something other than ResourceFile (meaning can a user get a `resource` in `values` that isn't a `ResourceFile`?

Reply:
No not as currently implemented. @danking What do you think about asserts? I think they're helpful so I remember what the type of the input should be. Does Python have the equivalent of `require`?

Reply:
I think the assertions are helpful. I don't see what the benefit is to taking them out and they'll help pinpoint bugs closer to where they originated.

Reply:
The benefit is that you don't need to place asserts everywhere. If there is something that the user is passing in that could be not a ResourceFile, I totally agree with you, we want to catch the error early and tell them. If this is a runtime check to make sure we didn't change our internal API, I think that should be a test of this function; it will fail at `resource.add_resource_group(self)`.